src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -507,17 +511,33 @@ private List<GenericKubernetesResource> getBindableKinds() {
     }
 
     @Override
-    public List<ServiceTemplate> getServiceTemplates() throws IOException {
+    public List<ServiceTemplate> getServiceTemplates() {
         try {
             List<GenericKubernetesResource> bindableKinds = getBindableKinds();
-            OpenShiftOperatorHubAPIGroupDSL hubClient = new OpenShiftOperatorHubAPIGroupClient(client);
+            OpenShiftClient osClient = toOpenShiftClient(client);
+            if (osClient == null) {
+                return Collections.emptyList();
+            }
+            OpenShiftOperatorHubAPIGroupDSL hubClient = osClient.operatorHub();
             ServiceTemplatesDeserializer deserializer = new ServiceTemplatesDeserializer(this::findSchema, bindableKinds);
             return deserializer.fromList(hubClient.clusterServiceVersions().list());
         } catch (KubernetesClientException e) {
             return Collections.emptyList();
         }
     }
 
+    private OpenShiftClient toOpenShiftClient(KubernetesClient client) {
+        OpenShiftClient osClient = client.adapt(OpenShiftClient.class);
+        try {
+            if (osClient.isSupported()) {
+                return osClient;
+            }
+        } catch (KubernetesClientException e) {
+            // unauthorized
```

Comment:
ideally we should throw this error and catch where we can display the error that the user is not authorized.